src/CMakeLists.txt

```diff
@@ -49,6 +54,23 @@ foreach(LIBRARY shadow-shim-helper-rs logger shadow-shmem shadow-tsc shadow-rs)
   set_target_properties(${LIBRARY} PROPERTIES IMPORTED_LOCATION_RELEASE ${TARGETDIR}/release/${LIBRARY_FILE})
 endforeach()
 
+## Unit test executables don't have predictable names: https://github.com/rust-lang/cargo/issues/1924
+##
+## We previously looked for executables with names matching '${UNDERBARRED_CRATENAME}*' in
+## `${TARGETDIR}/${RUST_BUILD_TYPE}/deps/`, but this would require us to enumerate all the crates
+## with unit tests, which would be particularly error prone now that we're not individually building
+## the crates from cmake. (This is a larger set than the library-list above, which are only
+## the crates that we need to link from C code)
+##
+## Just using `cargo test` for now. It potentially (re)builds code, but seems
+## better than trying to # fight the "rust way" here. It also includes doc tests,
+## which normally don't have persistent executables at all.
+add_test(NAME rust-unit-tests
+         COMMAND bash -c "${CARGO_ENV_VARS} cargo test ${RUST_BUILD_FLAG} --manifest-path \"${CMAKE_CURRENT_SOURCE_DIR}\"/Cargo.toml --target-dir \"${TARGETDIR}\" --features \"${RUST_FEATURES}\"")
+## Longer timeout here and run serially, since it's running a whole test suite,
+#and may end up rebuilding code # if anything has changed.
```

Comment:
There are two spurious `#` characters in the comments here.

Reply:
Oops, I think this was at least partly an IDE auto-re-wrapping error; it doesn't seem to like the double ## leaders. Changed them to a single #.